src/main/java/com/mecatran/gtfsvtor/validation/dao/DropOffPickUpTypeForSplitOrJoinedTripsValidator.java

```diff
@@ -0,0 +1,179 @@
+package com.mecatran.gtfsvtor.validation.dao;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimaps;
+import com.mecatran.gtfsvtor.dao.IndexedReadOnlyDao;
+import com.mecatran.gtfsvtor.model.GtfsDropoffType;
+import com.mecatran.gtfsvtor.model.GtfsPickupType;
+import com.mecatran.gtfsvtor.model.GtfsStopTime;
+import com.mecatran.gtfsvtor.model.GtfsTrip;
+import com.mecatran.gtfsvtor.reporting.ReportSink;
+import com.mecatran.gtfsvtor.reporting.issues.WrongDropOffPickUpTypeForSplitTripsIssue;
+import com.mecatran.gtfsvtor.validation.DaoValidator;
+
+public class DropOffPickUpTypeForSplitOrJoinedTripsValidator
+		implements DaoValidator {
+
+	/**
+	 * Validates that boarding restrictions for split and joined trains are
+	 * correctly set.
+	 * Boarding restrictions (pickup / drop off type) are set correctly, if
+	 * for the stops in common only one of these trips allows pickup (in case
+	 * of joined trains) or drop-off (in case of split trains).
+	 * <p>
+	 * Note that this validation assumes, that split/joined trips have the same service id.
+	 *
+	 * @param context validation context
+	 */
+	@Override
+	public void validate(Context context) {
+		IndexedReadOnlyDao dao = context.getDao();
+		ReportSink reportSink = context.getReportSink();
+
+		// Process trips route by route to reduce memory usage
+		// split trains have at least their two first stops in common, joined
+		// at least their last stops. We assume that both trips have the same
+		// service id. That means stop id, departure and arrival time
+		// match so we collect all split/joined trips by a key representing
+		// their first/last to stops and service id.
+		dao.getRoutes().forEach(route -> {
```

Comment:
Should not we check for route type and restrict to a (configurable?) subset of types? Joining and splitting trains make sense for rail mode, but maybe for buses this situation can be legitimate.

Reply:
Indeed. I added a ConfigurableOption. Not sure if configuration should better only support primitives, but parsin route_types directly could be of value for other validators as well